### ISSUE TITLE
Usagi with non-Oracle JDKs?

### ISSUE THREAD

**Issue by quinntd** [OHDSI/Usagi#103]
Cross-posting from:  https://forums.ohdsi.org/t/does-usagi-require-oracle-java-1-8-can-i-use-openjdk-or-amazon-corretto/13171

We’re trying to install Usagi on a multi-user Windows server on which most users do not have admin privileges.

Because of Oracle’s new (unfortunate) stance on licensing the Java Platform Standard Edition in production environments, we only have OpenJDK 12 and Amazon Corretto 8 installed on this Windows server.

Apparently, Usagi doesn’t like this. It fails to parse our Athena files into the required BerkeleyDB and Apache Lucene files (in the sleepyCat and mainIndex directories, respectively). It hangs on the initial dialog box warning us that this one-time indexing process may take a while. (We let it run for more than 48 hours before killing it.)

In comparison, Usagi indexed our Athena files in about 20 minutes running locally on a laptop with Oracle Java 8 (version 1.8) installed.

Any chance Usagi can support a non-Oracle JDK?  Or must we always consider our usage of Usagi to be non-commercial?

**Comment by quinntd** [comment]
A status update on this problem:

It wasn't the non-Oracle Java JDK causing problems; it was the permissions of the Usagi `.jar` file's directory on our multi-user Windows server.

We installed Usagi in `C:\Program Files\Usagi`, which is accessible to all users of the server but to which individual users did not have the ability to create new files or sub-directories.  Consequently, Usagi's attempted indexing of the Athena vocabulary files was failing.

Updating the Usagi directory's permissions to allow file & directory creation did the trick.  However, counter-intuitively, the `mainIndex` and `sleepyCat` sub-directories were created in my `C:\Users\myusername` directory instead of in `C:\Program Files\Usagi`.  Usagi didn't know where to go to access them.

I moved these sub-directories and the four `.txt` text files over to the `C:\Program Files\Usagi` directory and Usagi now sees the indexed Athena data.

In summary, Usagi worked with our non-Oracle Java JDK.  But you could consider refining some of the mechanics of directory and file locations.

**Comment by MaximMoinat** [comment]
Thanks for the feedback, happy to read you were able to solve the problem in the end.

I have seen the issue with Usagi using a different folder to create the indexes before. I will create a separate issue for that.
